src/results.cpp

```diff
@@ -484,9 +481,8 @@ TableView Results::get_tableview()
     REALM_UNREACHABLE();
 }
 
-Results Results::sort(realm::SortOrder&& sort) const
+Results Results::sort(realm::SortDescriptor&& sort) const
 {
-    REALM_ASSERT(sort.column_indices.size() == sort.ascending.size());
```

Comment:
That should probably be done *somewhere*, but the SortDescriptor currently doesn't actually know what table it's for. Maybe I can make it check on use at least.